### ISSUE TITLE
sitemapindex with child-sitemaps returns always the same sitemap

### ISSUE THREAD

**Issue by oripka** [d3xter-dev/sitemap-module-nuxt-3#4]
Using the following configuration

```
const defaultOptionsExample = {
    hostname: 'http://localhost:3000',
    exclude: ['/**'],
    //gzip: true
}


export const sitemap = async () =>  {
    return {
        exclude: ['/**'],
        path: '/sitemapindex.xml',
        sitemaps : [
            {
                ...defaultOptionsExample,
                path: `sitemap-1.xml`,
                routes: ["/bla", "/fasel"],
            },
            {
                ...defaultOptionsExample,
                path: `sitemap-2.xml`,
                routes: ["/xxx", "/yyy"],
            }
        ]
    }
}
```

I get this for `sitemap-1.xml`

```<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:mobile="http://www.google.com/schemas/sitemap-mobile/1.0" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
<url>
<loc>https://localhost:3000/bla</loc>
</url>
<url>
<loc>https://localhost:3000/fasel</loc>
</url>
</urlset>
```

and the exact same content for `sitemap-2.xml`

```<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:mobile="http://www.google.com/schemas/sitemap-mobile/1.0" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
<url>
<loc>http://localhost:3000/bla</loc>
</url>
<url>
<loc>http://localhost:3000/fasel</loc>
</url>
</urlset>
```

If I load `sitemap-2.xml` first I get for both sitemaps.

```
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:mobile="http://www.google.com/schemas/sitemap-mobile/1.0" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
<url>
<loc>http://localhost:3000/xxx</loc>
</url>
<url>
<loc>http://localhost:3000/yyy</loc>
</url>
</urlset>
```

It seems like the first content that gets generated is also show as the other sitemaps or some kind of caching issue.

**Comment by d3xter-dev** [comment]
Hey sorry for the late response! 
Would be nice if you could provide a  Vitest test,  would save me some time.     
Examples can be found in https://github.com/funkenstudio/sitemap-module-nuxt-3/blob/experimental/test/module.test.js 

Even if not  I will still try to look at this start of next week :)

**Comment by oripka** [comment]
Reading the mentioned file I saw that the vitests for my issue are already implemented.

On my system they all fail. See this screenshot. Comparing these tests to the behavior I see in my app it is actually weird, because at least the test for `sitemapindex.xml` should succeed. I am wondering if somehow my setup is causing this (i am running all nuxt within docker).

```
bash-5.1# yarn nuxt-build-module && yarn test
yarn run v1.22.19
$ /app/data/sitemap-module-nuxt-3/node_modules/.bin/nuxt-build-module
ℹ Building @funken-studio/sitemap-nuxt-3                                                                                                      08:11:51
✔ Build succeed for @funken-studio/sitemap-nuxt-3                                                                                             08:12:01
  /app/data/sitemap-module-nuxt-3/dist/runtime (chunks: 16 files)                                                                             08:12:01
  dist/module.mjs (size: 69.6 kB, exports: default)                                                                                           08:12:01
                                                                                                                                              08:12:01
Done in 12.24s.
yarn run v1.22.19
$ yarn lint && yarn unit
$ eslint . --ext js,vue
$ vitest run

 RUN  v0.15.2 /app/data/sitemap-module-nuxt-3

```

![image](https://user-images.githubusercontent.com/227032/183601074-153b01d0-6f3f-4361-9476-f8125af1413e.png)

![image](https://user-images.githubusercontent.com/227032/183601417-7c720ea2-63c6-42cb-a1bb-b8d00e13fcde.png)

**Comment by tiendatleo** [comment]
I have the same issue

**Comment by lyoko515** [comment]
I have the same issue

**Comment by d3xter-dev** [comment]
Should be finally fixed with version 4.0.4
